tests/components/honeywell/test_init.py

```diff
@@ -48,3 +56,25 @@ async def test_setup_multiple_thermostats_with_same_deviceid(
     assert config_entry.state is ConfigEntryState.LOADED
     assert hass.states.async_entity_ids_count() == 1
     assert "Platform honeywell does not generate unique IDs" not in caplog.text
+
+
+async def test_away_temps_migration(hass: HomeAssistant) -> None:
+    """Test away temps migrate to config options."""
+    legacy_config = MockConfigEntry(
+        domain=DOMAIN,
+        data={
+            CONF_USERNAME: "fake",
+            CONF_PASSWORD: "user",
+            CONF_COOL_AWAY_TEMPERATURE: 1,
+            CONF_HEAT_AWAY_TEMPERATURE: 2,
+        },
+        options={},
+    )
+
+    legacy_config.add_to_hass(hass)
+    await hass.config_entries.async_setup(legacy_config.entry_id)
```

Comment:
We need to patch the library as well so it doesn't try to do I/O

Reply:
Done.